homeassistant/components/generic_thermostat/config_flow.py

```diff
@@ -0,0 +1,119 @@
+"""Config flow for the Generic Thermostat integration."""
+from __future__ import annotations
+
+import voluptuous as vol
+
+from homeassistant.components.climate import HVACMode
+from homeassistant.components.sensor import SensorDeviceClass
+from homeassistant.config_entries import ConfigEntry, ConfigFlow
+from homeassistant.const import PRECISION_HALVES, PRECISION_TENTHS, PRECISION_WHOLE
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import selector
+from homeassistant.helpers.schema_config_entry_flow import (
+    SchemaFlowFormStep,
+    SchemaOptionsFlowHandler,
+)
+
+from .const import (
+    CONF_AC_MODE,
+    CONF_COLD_TOLERANCE,
+    CONF_HEATER,
+    CONF_HOT_TOLERANCE,
+    CONF_INITIAL_HVAC_MODE,
+    CONF_KEEP_ALIVE,
+    CONF_MAX_TEMP,
+    CONF_MIN_DUR,
+    CONF_MIN_TEMP,
+    CONF_NAME,
+    CONF_PRECISION,
+    CONF_SENSOR,
+    CONF_TARGET_TEMP,
+    CONF_TEMP_STEP,
+    DEFAULT_MIN_TEMP,
+    DEFAULT_NAME,
+    DEFAULT_TOLERANCE,
+    DOMAIN,
+)
+
+CONF_PRECISION_LIST = [
+    selector.SelectOptionDict(value=str(PRECISION_TENTHS), label=str(PRECISION_TENTHS)),
+    selector.SelectOptionDict(value=str(PRECISION_HALVES), label=str(PRECISION_HALVES)),
+    selector.SelectOptionDict(value=str(PRECISION_WHOLE), label=str(PRECISION_WHOLE)),
+]
+
+OPTIONS_SCHEMA = vol.Schema(
+    {
+        vol.Optional(CONF_AC_MODE): bool,
+        vol.Optional(CONF_MAX_TEMP): float,
```

Comment:
Use a `NumberSelector` with precision set to "any" and mode set to "box" for the floats